es6-wrapped-promise.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+module.exports = (Promise, ensureAslWrapper) => {
+  // Updates to this class should also be applied to the the ES3 version
+  // in index.js.
+  return class WrappedPromise extends Promise {
+    constructor(executor) {
+      var context, args;
+      super(wrappedExecutor);
+      var promise = this;
+
+      try {
+        executor.apply(context, args);
+      } catch (err) {
+        args[1](err)
```

Comment:
I'd say it's a mistake that there were no semicolon in the ES3 counter part. We should probably fix that as well 😃